content/200-orm/200-prisma-client/500-deployment/301-edge/485-deploy-to-vercel.mdx

```diff
@@ -0,0 +1,458 @@
+
+First, ensure that the `DATABASE_URL` is set as the `url` of the `datasource` in your Prisma schema:
+
+```prisma
+datasource db {
+  provider = "postgresql" // this might also be `mysql` or another value depending on your database
+  url      = env("DATABASE_URL")
+}
+```
+
+#### Development
+
+When in **development**, you can configure your database connection via the `DATABASE_URL` environment variable (e.g. [using `.env` files](/orm/more/development-environment/environment-variables/env-files)).
+
+#### Production
+
+When deploying your Edge Function to **production**, you'll need to set the database connection using the `vercel` CLI:
+
+```terminal
+npx vercel env add DATABASE_URL
+```
+
+This command is interactive and will ask you to select environments and provide the value for the `DATABASE_URL` in subsequent steps.
+
+Alternatively, you can configure the environment variable [via the UI](https://vercel.com/docs/projects/environment-variables#declare-an-environment-variable) of your project in the Vercel Dashboard.
+
+### Generate Prisma Client in `postinstall` hook
+
+In your `package.json`, you should add a `"postinstall"` section as follows:
+
+```js file=package.json
+{
+  // ...,
+  "postinstall: "prisma generate"
+}
+```
+
+### Size limits on free accounts
+
+Vercel has a [size limit of 1 MB on free accounts](https://vercel.com/docs/functions/limitations). If your application bundle with Prisma ORM exceeds that size, we recommend upgrading to a paid account or using Prisma Accelerate to deploy your application.
+
+## Database-specific considerations & examples
+
+This section provides database-specific instructions for deploying a Vercel Edge Functions with Prisma ORM.
+
+### Prerequisites
+
+As a prerequisite for the following section, you need to have a Vercel Edge Function (which typically comes in the form of a Next.js API route) running locally and the Prisma and Vercel CLIs installed. If you don't have that yet, you can run these commands to set up a Next.js app from scratch (following the instructions of the [Vercel Edge Quickstart](https://vercel.com/docs/functions/edge-functions/quickstart)):
+
+```terminal
+npm install -g vercel
+npx create-next-app@latest
+npm install prisma --save-dev
+npx prisma init
+```
+
+We'll use the default `User` model for the example below:
+
+```prisma
+model User {
+  id    Int     @id @default(autoincrement())
+  email String  @unique
+  name  String?
+}
+```
+
+### Vercel Postgres
+
+If you are using Vercel Postres, you need to:
+
+- use the `@prisma/adapter-neon` database adapter (via the `driverAdapters` Preview feature) because Vercel Postgres uses [Neon](https://neon.tech/) under the hood
+- be aware that Vercel by default calls the environment variable for the database connection string `POSTGRES_PRISMA_URL` while the default name used in the Prisma docs is typically `DATABASE_URL`; using Vercel's naming, you need to set the following fields on your `datasource` block:
```

Comment:
This has 6 impact on Prisma Client, why should the user make this change?

Reply:
> Is the 6 a typo in there or what's that supposed to mean?
Yes, typo.

Reply:
> I opted for using the Vercel's naming of the env vars in the sample instructions because that allows for easier copying and pasting the values from the Vercel Dashboard. Do you think the sample instructions should use DATABASE_URL and DIRECT_URL instead or have another suggestion?

That depends on what you want to achieve.

On deployment Vercel just sets the define env var value with the connection string of Vercel Postgres, the user does not have to do anything about it. If you want to use `DATABASE_URL`, you as the user need to actually copy/paste the value manually and set it as an env var in Vercel for it to work.

Locally during development on the other hand, the user will probably have to do that via `.env` for the database connection string to be available.

So this needs to be consistent and hopefully work both in development and deployment - because driver adapters need to get the environment values from _code_, which means that of course one can not use different ones in dev and deploy.

Reply:
I tested this and it worked both in dev and prod like this.